From 1e5fad6443217f1988acb7298efc7fc0c6cfb726 Mon Sep 17 00:00:00 2001
From: Peter Vary <pvary@cloudera.com>
Date: Mon, 9 Oct 2017 10:29:52 +0200
Subject: [PATCH 1303/1363] CDH-60171 HIVE-17633: Make it possible to override
 the query results directory in TestBeeLineDriver
 (Peter Vary reviewed by Barna Zsombor Klara)

(cherry picked from commit 75c31e7ab232796511875c2aafad6086b793a4f5)

Change-Id: Icf2bb03a79ef0c6f2e79d5b08c13afd083fdf550
---
 data/scripts/q_test_init.sql                       |   10 +++++-----
 .../hadoop/hive/cli/control/CoreBeeLineDriver.java |   14 ++++++++++++--
 2 files changed, 17 insertions(+), 7 deletions(-)

diff --git a/data/scripts/q_test_init.sql b/data/scripts/q_test_init.sql
index 0c8668c..7c2b692 100644
--- a/data/scripts/q_test_init.sql
+++ b/data/scripts/q_test_init.sql
@@ -262,9 +262,9 @@ create table cbo_t1(key string, value string, c_int int, c_float float, c_boolea
 create table cbo_t2(key string, value string, c_int int, c_float float, c_boolean boolean)  partitioned by (dt string) row format delimited fields terminated by ',' STORED AS TEXTFILE;
 create table cbo_t3(key string, value string, c_int int, c_float float, c_boolean boolean)  row format delimited fields terminated by ',' STORED AS TEXTFILE;
 
-load data local inpath '../../data/files/cbo_t1.txt' into table cbo_t1 partition (dt='2014');
-load data local inpath '../../data/files/cbo_t2.txt' into table cbo_t2 partition (dt='2014');
-load data local inpath '../../data/files/cbo_t3.txt' into table cbo_t3;
+load data local inpath '${hiveconf:test.data.dir}/cbo_t1.txt' into table cbo_t1 partition (dt='2014');
+load data local inpath '${hiveconf:test.data.dir}/cbo_t2.txt' into table cbo_t2 partition (dt='2014');
+load data local inpath '${hiveconf:test.data.dir}/cbo_t3.txt' into table cbo_t3;
 
 CREATE TABLE part(
     p_partkey INT,
@@ -278,7 +278,7 @@ CREATE TABLE part(
     p_comment STRING
 );
 
-LOAD DATA LOCAL INPATH '../../data/files/part_tiny.txt' overwrite into table part;
+LOAD DATA LOCAL INPATH '${hiveconf:test.data.dir}/part_tiny.txt' overwrite into table part;
 
 CREATE TABLE lineitem (L_ORDERKEY      INT,
                                 L_PARTKEY       INT,
@@ -299,7 +299,7 @@ CREATE TABLE lineitem (L_ORDERKEY      INT,
 ROW FORMAT DELIMITED
 FIELDS TERMINATED BY '|';
 
-LOAD DATA LOCAL INPATH '../../data/files/lineitem.txt' OVERWRITE INTO TABLE lineitem;
+LOAD DATA LOCAL INPATH '${hiveconf:test.data.dir}/lineitem.txt' OVERWRITE INTO TABLE lineitem;
 
 create table src_cbo as select * from src;
 
diff --git a/itests/util/src/main/java/org/apache/hadoop/hive/cli/control/CoreBeeLineDriver.java b/itests/util/src/main/java/org/apache/hadoop/hive/cli/control/CoreBeeLineDriver.java
index 1fdce17..b44ffbd 100644
--- a/itests/util/src/main/java/org/apache/hadoop/hive/cli/control/CoreBeeLineDriver.java
+++ b/itests/util/src/main/java/org/apache/hadoop/hive/cli/control/CoreBeeLineDriver.java
@@ -58,7 +58,12 @@ public CoreBeeLineDriver(AbstractCliConfig testCliConfig) {
     super(testCliConfig);
     queryDirectory = new File(testCliConfig.getQueryDirectory());
     logDirectory = new File(testCliConfig.getLogDir());
-    resultsDirectory = new File(testCliConfig.getResultsDir());
+    String testResultsDirectoryName = System.getProperty("test.results.dir");
+    if (testResultsDirectoryName != null) {
+      resultsDirectory = new File(hiveRootDirectory, testResultsDirectoryName);
+    } else {
+      resultsDirectory = new File(testCliConfig.getResultsDir());
+    }
     String testDataDirectoryName = System.getProperty("test.data.dir");
     if (testDataDirectoryName == null) {
       testDataDirectory = new File(hiveRootDirectory, "data" + File.separator + "files");
@@ -66,7 +71,12 @@ public CoreBeeLineDriver(AbstractCliConfig testCliConfig) {
       testDataDirectory = new File(testDataDirectoryName);
     }
     testScriptDirectory = new File(hiveRootDirectory, "data" + File.separator + "scripts");
-    initScript = new File(testScriptDirectory, testCliConfig.getInitScript());
+    String initScriptFileName = System.getProperty("test.init.script");
+    if (initScriptFileName != null) {
+      initScript = new File(testScriptDirectory, initScriptFileName);
+    } else {
+      initScript = new File(testScriptDirectory, testCliConfig.getInitScript());
+    }
     cleanupScript = new File(testScriptDirectory, testCliConfig.getCleanupScript());
   }
 
-- 
1.7.9.5

